### PR TITLE
Add mentoring combo button

### PR DESCRIPTION
**Pull request by kntsoriano**
## Description
This adds the mentoring combo button to the <SolutionSummary /> component. 

@iHiD I'm pretty sure this contains N+1s. Could you check that? Also, I tried to pass in the data to the react component, but I'm unsure if it's correct.

### CODE REVIEW

File: app/javascript/components/common/CopyToClipboardButton.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ export function CopyToClipboardButton({ textToCopy }: { textToCopy: string }) {
       className={`c-copy-text-to-clipboard ${justCopied ? 'copied' : ''}`}
       aria-label={`Copy ${textToCopy} to the cliboard`}
     >
-      <div className="text">exercism download --exercise=pattern</div>
+      <div className="text">{textToCopy}</div>
```
Comment (kntsoriano) [inline app/javascript/components/common/CopyToClipboardButton.tsx:77]: @ErikSchierboom I noticed that this widget doesn't show the correct text to copy.

File: app/javascript/components/student/solution-summary/Mentoring.tsx [hunk 2]
```diff
@@ -10,16 +28,115 @@ export const Mentoring = ({ link }: { link: string }): JSX.Element => {
...
+  styles: { [key: string]: React.CSSProperties }
+  attributes: {
+    [key: string]: {
+      [key: string]: string
+    }
+  }
+  setPanelElement: React.Dispatch<React.SetStateAction<HTMLDivElement | null>>
+}
+
+const MentoringPanel = ({
```
Comment (kntsoriano) [inline app/javascript/components/student/solution-summary/Mentoring.tsx:123]: I render the dropdown panel in a portal in order for it to not inherit the styles from the parent div. We should probably do this for all of our panels moving forward.

File: app/javascript/components/common/CopyToClipboardButton.tsx (hunk 1 above)
Comment (ErikSchierboom) [reply app/javascript/components/common/CopyToClipboardButton.tsx:77]: Ah, silly!

File: app/helpers/react_components/student/solution_summary.rb [hunk 3]
```diff
@@ -34,9 +35,30 @@ def links
           community_solutions: "#",
           learn_more_about_mentoring_article: "#",
           mentoring_info: "#",
-          complete_exercise: Exercism::Routes.complete_api_solution_url(solution.uuid)
+          complete_exercise: Exercism::Routes.complete_api_solution_url(solution.uuid),
+          share_mentoring: "https://some.link/we/need/to-decide-on",
```
Comment (ErikSchierboom) [inline app/helpers/react_components/student/solution_summary.rb]: Could this have the text TODO in it somewhere, to be able to find it later? Or do you expect to update it in this PR?

**Review by ErikSchierboom — APPROVED** [review]

File: app/javascript/components/common/CopyToClipboardButton.tsx (hunk 1 above)
Comment (iHiD) [reply app/javascript/components/common/CopyToClipboardButton.tsx:77]: Ha. I fixed this in https://github.com/exercism/website/pull/573/files too :)

File: app/helpers/react_components/student/solution_summary.rb (hunk 3 above)
Comment (iHiD) [reply app/helpers/react_components/student/solution_summary.rb]:
```suggestion
          share_mentoring: "https://some.link/we/need/to-decide-on", # TODO
```

File: app/helpers/react_components/student/solution_summary.rb [hunk 4]
```diff
@@ -34,9 +35,30 @@ def links
...
+          request_mentoring: "#"
```
Comment (iHiD) [inline app/helpers/react_components/student/solution_summary.rb]:
```suggestion
          request_mentoring: "#" # TODO
```

File: app/javascript/components/student/solution-summary/Mentoring.tsx (hunk 2 above)
Comment (iHiD) [reply app/javascript/components/student/solution-summary/Mentoring.tsx:123]: That's a good idea. Does it matter from an a11y POV that the focus switches to something that's far away in the DOM?

Comment (kntsoriano) [reply app/javascript/components/student/solution-summary/Mentoring.tsx:123]: I'm not entirely sure. I'll look into it.

**Pull request merged**